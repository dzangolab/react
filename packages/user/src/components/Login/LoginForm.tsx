import {
  Email,
  Form,
  Password,
  emailSchema,
  passwordSchema,
} from "@dzangolab/react-form";
import { useTranslation } from "@dzangolab/react-i18n";
import { SubmitButton } from "@dzangolab/react-ui";
import * as zod from "zod";

import { LoginCredentials } from "../../types";

interface Properties {
  handleSubmit: (credentials: LoginCredentials) => void;
  loading?: boolean;
}

export const LoginForm = ({ handleSubmit, loading }: Properties) => {
  const { t } = useTranslation("user");

  const LoginFormSchema = zod.object({
    email: emailSchema({
      invalid: t("validation.messages.validEmail"),
      required: t("validation.messages.email"),
    }),
    password: passwordSchema(
      {
        required: t("login.messages.validation.password"),
        weak: "",
      },
      {
        minLength: 0,
      },
    ),
  });

  return (
    <Form validationSchema={LoginFormSchema} onSubmit={handleSubmit}>
      <Email
        label={t("login.form.email.label")}
        name="email"
        placeholder={t("login.form.email.placeholder")}
        showValidState={false}
      />
      <Password
        label={t("login.form.password.label")}
        name="password"
        showValidState={false}
      />
      <SubmitButton label={t("login.form.actions.submit")} loading={loading} />
    </Form>
  );
};
