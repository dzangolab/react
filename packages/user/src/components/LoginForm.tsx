import { useTranslation } from "@dzangolab/react-i18n";
import { LoadingButton } from "@dzangolab/react-ui";
import { Field, Formik } from "formik";
import * as Yup from "yup";

import { LoginCredentials } from "@/types/types";

import "../assets/css/loginForm.css";
import ErrorMessage from "./ErrorMessage";

interface Properties {
  handleSubmit: (credentials: LoginCredentials) => void;
  loading?: boolean;
}

const LoginForm = ({ handleSubmit, loading }: Properties) => {
  const { t } = useTranslation("user");

  const LoginFormSchema = Yup.object({
    email: Yup.string()
      .email("validation.messages.validEmail")
      .required("validation.messages.email"),
    password: Yup.string().required("login.messages.validation.password"),
  });

  const initialValue = {
    email: "",
    password: "",
  };

  return (
    <Formik
      initialValues={initialValue}
      validationSchema={LoginFormSchema}
      onSubmit={(values, action) => {
        handleSubmit(values);
        action.resetForm();
      }}
    >
      {({ errors, handleSubmit, touched }) => (
        <form className="form" onSubmit={handleSubmit}>
          <div className="field email">
            <label htmlFor="email">{t("login.form.email.label")}</label>
            <Field
              id="email"
              type="email"
              name="email"
              placeholder={t("login.form.email.placeholder")}
            />
            <ErrorMessage
              touched={touched.email}
              error={errors.email ? t(errors.email) : undefined}
            />
          </div>

          <div className="field password">
            <label htmlFor="password">{t("login.form.password.label")}</label>
            <Field id="password" type="password" name="password" />
            <ErrorMessage
              touched={touched.password}
              error={errors.password ? t(errors.password) : undefined}
            />
          </div>

          <div className="actions">
            <LoadingButton
              label={`${t("login.form.actions.submit")}`}
              loading={loading}
            />
          </div>
        </form>
      )}
    </Formik>
  );
};

export default LoginForm;
