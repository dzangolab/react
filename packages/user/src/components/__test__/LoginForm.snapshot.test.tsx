import React from "react";
import { create } from "react-test-renderer";
import { expect, test } from "vitest";

import LoginForm from "../LoginForm";

import type {
  ReactTestRenderer,
  ReactTestRendererJSON,
} from "react-test-renderer";

function toJson(component: ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const component = create(<LoginForm handleSubmit={() => {}} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
