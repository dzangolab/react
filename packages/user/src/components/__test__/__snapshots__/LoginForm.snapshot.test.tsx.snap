// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="field-wrapper undefined"
  >
    <label
      htmlFor="email"
    >
      login.form.email.label
    </label>
    <input
      className="input-field email"
      disabled={false}
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="login.form.email.placeholder"
      readOnly={false}
      type="email"
      value=""
    />
  </div>
  <div
    className="field password"
  >
    <label
      htmlFor="input-field-password"
    >
      login.form.password.label
    </label>
    <div
      className="input-field-password"
    >
      <input
        id="input-field-password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        type="password"
      />
      <span
        className="eye-icon"
        onClick={[Function]}
      >
        <i
          className="pi pi-eye-slash"
        />
      </span>
    </div>
  </div>
  <div
    className="dz-form-actions"
    data-alignment="fill"
    data-direction="horizontal"
  >
    <button
      className="dz-button primary filled medium"
      id="submit"
      role="button"
      type="submit"
    >
      <span
        className="label"
      >
        login.form.actions.submit
      </span>
    </button>
  </div>
</form>
`;
